content/300-guides/400-migrate-to-prisma/01-migrate-from-typeorm.mdx

```diff
@@ -1078,3 +1078,9 @@ model Post {
 ```
 
 This would also result in a more ergonomic and less verbose Prisma Client API to modify the records in this relation, because you have a direct path from `Post` to `Category` (and the other way around) instead of needing to traverse the `PostToCategories` model first.
+
+<Admonition type="warning">
+  
+If your database provider automatically creates a primary key, and you want to use m-n relations, you need to use explicit relation syntax, manually create the join model, and verify that this join model has a primary key. This is because Join tables created by Prisma for m-n relations using implicit syntax in Prisma Schema do not have primary keys. 
```

Comment:
I may be wrong but I think 

>If your database provider automatically creates a primary key,

should be

>If your database provider requires tables to have primary keys,

Some providers like DigitalOcean enforce primary keys on all tables for replication. In such a case, the user would have to switch to the explicit syntax.